sa/sa_test.go

```diff
@@ -4565,3 +4567,346 @@ func TestUpdateRegistrationKey(t *testing.T) {
 		})
 	}
 }
+
+type mockRLOStream struct {
+	grpc.ServerStream
+	sent []*sapb.RateLimitOverride
+	ctx  context.Context
+}
+
+func newMockRLOStream() *mockRLOStream {
+	return &mockRLOStream{ctx: ctx}
+}
+func (m *mockRLOStream) Context() context.Context { return m.ctx }
+func (m *mockRLOStream) RecvMsg(any) error        { return io.EOF }
+func (m *mockRLOStream) Send(ov *sapb.RateLimitOverride) error {
+	m.sent = append(m.sent, ov)
+	return nil
+}
+
+func TestAddRateLimitOverrideInsertThenUpdate(t *testing.T) {
+	if os.Getenv("BOULDER_CONFIG_DIR") != "test/config-next" {
+		t.Skip("skipping, this overrides table must exist for this test to run")
+	}
+
+	sa, _, cleanup := initSA(t)
+	defer cleanup()
+
+	expectBucketKey := core.RandomString(10)
+	ov := &sapb.RateLimitOverride{
+		LimitEnum: 1,
+		BucketKey: expectBucketKey,
+		Comment:   "insert",
+		Period:    durationpb.New(time.Hour),
+		Count:     100,
+		Burst:     100,
+	}
+
+	// Insert
+	resp, err := sa.AddRateLimitOverride(ctx, &sapb.AddRateLimitOverrideRequest{Override: ov})
+	test.AssertNotError(t, err, "expected successful insert, got error")
+	test.Assert(t, resp.Inserted && resp.Enabled, fmt.Sprintf("expected (Inserted=true, Enabled=true) for initial insert, got (%v,%v)", resp.Inserted, resp.Enabled))
+
+	// Update (change comment)
+	ov.Comment = "updated"
+	resp, err = sa.AddRateLimitOverride(ctx, &sapb.AddRateLimitOverrideRequest{Override: ov})
+	test.AssertNotError(t, err, "expected successful update, got error")
+	test.Assert(t, !resp.Inserted && resp.Enabled, fmt.Sprintf("expected (Inserted=false, Enabled=true) for update, got (%v,%v)", resp.Inserted, resp.Enabled))
```

Comment:
Pedantic nit: add space between `(%v, %v)` to match the prior parenthetical

Reply:
🦅👁️